### PR TITLE
[dev] AB#25926 - [W.2] - Ability to create an entry for a connection form - Modification of default props on person panel react-cm-ui component 

### PR DESCRIPTION
**Pull request by fantaJinMode**
PR contains:
- Modification of default props for person panel component
- If `personId` isn't passed for `summaryData` then don't display personId

#note: @War777 I have been reverting some code which I believe was changed by you. Could you please verify it once?

### CODE REVIEW

File: src/dataDisplay/personPanel/personPanelDetails.jsx [hunk 1]
```diff
@@ -196,10 +196,7 @@ const defaultProps = {
     isExpanded: false,
     isMobile: false,
     otherDataGroups: null,
-    selectButtonProps: {
-        label: 'Select',
-        disable: false,
-    },
+    selectButtonProps: {},
```
Comment (groberts314) [inline src/dataDisplay/personPanel/personPanelDetails.jsx:199]: Is this change necessary?  I think we should keep it if possible to avoid regressions.  i.e. by default this button should show and have these props.  And someone (like us) that wishes to suppress this button should have to do something.

Maybe introduce a new `showSelectButton` prop (Boolean, optional, defaults to `true`, and we would set it to `false`).

File: src/dataDisplay/personPanel/personPanelSummary.jsx [hunk 2]
```diff
@@ -577,17 +577,19 @@ function PersonPanelSummary(props) {
...
-                    </Typography>
-                </Grid.Column>
+                        <Typography
+                            className={classes.personId}
+                            variant="caption"
+                        >
+                            {`Id: ${personId}`}
+                        </Typography>
+                    </Grid.Column>
+                )}
```
Comment (groberts314) [inline src/dataDisplay/personPanel/personPanelSummary.jsx:592]: This is not a bad enhancement, although we might double check if omission of Person ID on our comps (e.g. https://www.sketch.com/s/877c6c65-893c-456e-9d33-9bd002036677/a/Vrzbyeo ) is actually intentional.  We might want to show the Person ID, as this is helpful to our end-users I believe.  Let's double check with Sandra and Hannah at scrum today.

**Review by groberts314 — CHANGES_REQUESTED** [review]
I'm not quite sure the Select button stuff is quite what we should be doing.

In my opinion, let's not mess with the default value for `selectButtonProps`, and let's introduce a new prop `showSelectButton` (Boolean, optional, defaults to `true`, so we can then do `showSelectButton={false}` to suppress the Select button.

File: src/dataDisplay/personPanel/personPanelSummary.jsx (hunk 2 above)
Comment (fantaJinMode) [reply src/dataDisplay/personPanel/personPanelSummary.jsx:592]: This would also avoid situations like if personId isn't passed then it would show on UI `personId: undefined`

File: src/dataDisplay/personPanel/personPanelDetails.jsx (hunk 1 above)
Comment (fantaJinMode) [reply src/dataDisplay/personPanel/personPanelDetails.jsx:199]: @groberts314 Make sense if it's to avoid possible regressions, but it can be handled from UI as well. Should I add boolean `showSelectButton` and make those changes? Guys, what are your views? @War777 @cbrewer1689

Comment (cbrewer1689) [reply src/dataDisplay/personPanel/personPanelDetails.jsx:199]: I believe this change seems good. Having the default be `{}` makes sense here, because below, in the render, the label's value is being set to "Select" by default and `disable` is already be set to `false` inside of the PersonPanelDetailsActionButton component.

File: src/dataDisplay/personPanel/personPanelSummary.jsx (hunk 2 above)
Comment (cbrewer1689) [reply src/dataDisplay/personPanel/personPanelSummary.jsx:592]: If requirements say it is good, I am fine with this change.

**Review by cbrewer1689 — APPROVED** [review]

**Review by cbrewer1689 — CHANGES_REQUESTED** [review]
Mmm, thinking about it further, none of my thoughts have changed about the changes below, but I am asking for a requested change here to not rely on an empty selectButtonProps object to not render the button. I think we should have an extra flag, maybe `showSelectButton`.

File: src/dataDisplay/personPanel/personPanelDetails.jsx (hunk 1 above)
Comment (groberts314) [reply src/dataDisplay/personPanel/personPanelDetails.jsx:199]: So, it sounds like this is fine (change to default value for `selectButtonProps`), but let's also add `showSelectButton` prop as described to govern whether or not to render the button.

^^^ @fantaJinMode

File: src/dataDisplay/personPanel/personPanelSummary.jsx (hunk 2 above)
Comment (groberts314) [reply src/dataDisplay/personPanel/personPanelSummary.jsx:592]: Yeah, we brought this up during Team S.H.I.E.L.D. scrum this morning with Garrett and Hannah.  This is a good change for this component, but I do think we might bring back Person ID for our particular usage (and have it wrap to the next line).  You know how our end-users love their Person IDs! :-P

File: src/dataDisplay/personPanel/personPanelDetails.jsx (hunk 1 above)
Comment (fantaJinMode) [reply src/dataDisplay/personPanel/personPanelDetails.jsx:199]: If I am thinking correctly `showSelectButton` prop for `<PersonPanelDetails>` component, Yes ??
If that's the case, should I add another prop `showViewRecordButton` for another button as well ? @groberts314 @cbrewer1689

Comment (fantaJinMode) [reply src/dataDisplay/personPanel/personPanelDetails.jsx:199]: `showSelectButton` Should be prop for <PersonPanelDetails> ??
It would be something like this 
```
selectButtonProps: {
      showSelectButton: false,
  },
```
@groberts314 @cbrewer1689

Comment (groberts314) [reply src/dataDisplay/personPanel/personPanelDetails.jsx:199]: @fantaJinMode I'd say yes.  Please default the props to `true` so that by default nothing changes, and people can suppress the buttons by setting new props to `false`.

Comment (fantaJinMode) [reply src/dataDisplay/personPanel/personPanelDetails.jsx:203]: @groberts314 @cbrewer1689 Fixed. Please check once

Comment (groberts314) [reply src/dataDisplay/personPanel/personPanelDetails.jsx:203]: @fantaJinMode @cbrewer1689 Should we restore the prior default value for `selectButtonProps` or should we leave it as an empty object?

Comment (groberts314) [reply src/dataDisplay/personPanel/personPanelDetails.jsx:203]: I think if we're checking `!isEmpty(selectButtonProps)` below we want to restore the default value so that by default, the Select button is displayed, to avoid regressions.  Does that sound right?  Or am I missing something?

File: src/dataDisplay/personPanel/personPanelDetails.jsx [hunk 3]
```diff
@@ -891,28 +894,31 @@ function PersonPanelDetails(props) {
...
-                                    {...viewRecordButtonProps}
-                                    label="View Record"
-                                    outlined
-                                />
-                            )}
-                        </div>
-                    )}
+                    {
+                        ((!isEmpty(selectButtonProps) && showSelectButton) ||
+                        !isEmpty(viewRecordButtonProps)) && (
```
Comment (groberts314) [inline src/dataDisplay/personPanel/personPanelDetails.jsx:899]: Can we move some of these expressions out to variables calculated before the `return ( ... );`?
e.g.
```JavaScript
const shouldRenderSelectButton = showSelectButton && !isEmpty(selectButtonProps);
const shouldRenderViewButton = !isEmpty(viewRecordButtonProps);

// ...

return (
    // ...
    {(shouldRenderSelectButton || shouldRenderViewButton) && (
        <div
            className={classes.actions}
        >
             {shouldRenderSelectButton && (
                <PersonPanelDetailsActionButton
                    label="Select"
                    // eslint-disable-next-line react/jsx-props-no-spreading
                    {...selectButtonProps}
                />
             )}
             
             {shouldRenderViewButton && (
                <PersonPanelDetailsActionButton
                    // eslint-disable-next-line react/jsx-props-no-spreading
                    {...viewRecordButtonProps}
                    label="View Record"
                    outlined
                />
             )}
        </div>
    )}
);
```

**Comment by groberts314** [comment]
@fantaJinMode @cbrewer1689 I'm concerned about changing the default value of `selectButtonProps` causing a regression, and I also have an additional suggestion to help with the conditional button rendering logic to avoid computing certain conditions multiple times.  So my status is still at 'request changes'.

File: src/dataDisplay/personPanel/personPanelDetails.jsx (hunk 3 above)
Comment (fantaJinMode) [reply src/dataDisplay/personPanel/personPanelDetails.jsx]: Fixed

**Comment by fantaJinMode** [comment]
> @fantaJinMode @cbrewer1689 I'm concerned about changing the default value of `selectButtonProps` causing a regression, and I also have an additional suggestion to help with the conditional button rendering logic to avoid computing certain conditions multiple times. So my status is still at 'request changes'.

On default logic, waiting for @cbrewer1689 response and fixed conditional button rendering logic

File: src/dataDisplay/personPanel/personPanelDetails.jsx (hunk 1 above)
Comment (fantaJinMode) [reply src/dataDisplay/personPanel/personPanelDetails.jsx:203]: For avoiding regressions, I agree with @groberts314 of restoring the default value. Please provide your suggestion on this @cbrewer1689. According to that, I will make new changes

Comment (cbrewer1689) [reply src/dataDisplay/personPanel/personPanelDetails.jsx:203]: I think either way is good with me. I don't believe there would be any regressions with how this line current looks.

**Review by cbrewer1689 — APPROVED** [review]

File: src/dataDisplay/personPanel/personPanelDetails.jsx (hunk 1 above)
Comment (groberts314) [reply src/dataDisplay/personPanel/personPanelDetails.jsx:203]: @fantaJinMode I've confirmed with @cbrewer1689 and _really do_ need to revert this change on the default value for `selectButtonProps`.  Because otherwise, callers that are not setting this prop (intending to accept the default value) will have their buttons mysteriously disappear! (because of the `!isEmpty(selectButtonProps)` check).

Comment (groberts314) [reply src/dataDisplay/personPanel/personPanelDetails.jsx]: Looking good as of 2ae8cb3.  I consider this resolved, but since we all participated, I'll keep the thread visible for now.

File: src/dataDisplay/personPanel/personPanelDetails.jsx (hunk 3 above)
Comment (groberts314) [reply src/dataDisplay/personPanel/personPanelDetails.jsx]: This was resolved satisfactorily in c0008b4.

**Review by groberts314 — APPROVED** [review]

**Comment by groberts314** [comment]
@cbrewer1689 Any objections to merging this and kicking out a 10.1.6 release?
I am happy to do that if you like.
